packages/suite/src/components/suite/modals/confirm/ConfirmDeviceScreen.tsx

```diff
@@ -0,0 +1,117 @@
+import { TrezorDevice } from '@suite-types/index';
+import { Button, ConfirmOnDevice, ModalProps, variables } from '@trezor/components';
+import { getDeviceModel } from '@trezor/device-utils';
+import { copyToClipboard } from '@trezor/dom-utils';
+import DeviceDisconnected from './Address/components/DeviceDisconnected';
+
+const Wrapper = styled.div`
+    display: flex;
+    flex-direction: column;
+    align-self: center;
+`;
+
+const Value = styled.div`
+    font-size: ${variables.FONT_SIZE.NORMAL};
+    color: ${({ theme }) => theme.TYPE_DARK_GREY};
+    font-weight: ${variables.FONT_WEIGHT.MEDIUM};
+    font-variant-numeric: tabular-nums slashed-zero;
+    margin-bottom: 20px;
+    width: 100%;
+    background: ${({ theme }) => theme.BG_LIGHT_GREY};
+    border: 1px solid ${({ theme }) => theme.STROKE_GREY};
+    border-radius: 8px;
+    word-break: break-all;
+    padding: 10px;
+    max-width: calc(${QRCODE_SIZE}px + ${QRCODE_PADDING * 2}px);
+`;
+
+const CopyButtonWrapper = styled.div`
+    display: flex;
+    justify-content: center;
+`;
+
+const StyledModal = styled(Modal)`
+    width: unset;
+`;
+
+const StyledDeviceDisconnected = styled(DeviceDisconnected)`
+    max-width: calc(${QRCODE_SIZE}px + ${QRCODE_PADDING * 2}px);
+`;
+
+export interface ConfirmDeviceScreenProps
+    extends Pick<ModalProps, 'heading' | 'isCancelable' | 'onCancel'> {
+    device: TrezorDevice;
+    copyButtonText: React.ReactNode;
+    copyButtonDataTest?: string;
+    isConfirmed?: boolean;
+    value: string;
+    valueDataTest?: string;
+}
+
+export const ConfirmDeviceScreen = ({
+    copyButtonText,
+    copyButtonDataTest,
+    device,
+    heading,
+    isCancelable,
+    isConfirmed,
+    onCancel,
+    value,
+    valueDataTest,
+}: ConfirmDeviceScreenProps) => {
+    const htmlElement = useRef<HTMLDivElement>(null);
+
+    const dispatch = useDispatch();
+
+    const copy = () => {
+        const result = copyToClipboard(value, htmlElement.current);
+        if (typeof result !== 'string') {
+            dispatch(notificationsActions.addToast({ type: 'copy-to-clipboard' }));
+        }
+    };
+
+    return (
+        <StyledModal
+            heading={heading}
+            modalPrompt={
+                device.connected ? (
+                    <ConfirmOnDevice
+                        title={<Translation id="TR_CONFIRM_ON_TREZOR" />}
+                        deviceModel={getDeviceModel(device)}
+                        onCancel={isCancelable ? onCancel : undefined}
+                        isConfirmed={isConfirmed}
+                    />
+                ) : undefined
+            }
+            isCancelable={isCancelable}
+            onCancel={onCancel}
+        >
+            <Wrapper>
+                <QrCode value={value} />
+                <Value data-test={valueDataTest}>{value}</Value>
+                {device.connected ? (
+                    <CopyButtonWrapper ref={htmlElement}>
+                        {isConfirmed && (
```

Comment:
makes little sense to not show the copy button if the device is not connected IMO

Reply:
Consulted ondra and he agrees that the copy button can be rendered right away in case the device is not connected.

Reply:
Agreed, done.